pkg/querier/queryrange/split_by_interval.go

```diff
@@ -137,6 +148,10 @@ func (h *splitByInterval) Process(
 		}
 	}
 
+	if len(responses) == 0 && res != nil {
+		responses = append(responses, res)
+	}
```

Comment:
I think you could remove this `res` var and just append `&LokiResponse{}` here, that would also remove the extra if branch inside the code block above.

Reply:
![image](https://user-images.githubusercontent.com/9583245/127111569-714189e6-7727-4a30-a0c9-28539a07d8f9.png)
Because there are 4 types